### ISSUE TITLE
defn- should not be exported

### ISSUE THREAD

**Issue by aleedmanath** [squint-cljs/squint#455]
**Is your feature request related to a problem? Please describe.**

React HMR does not work as reliably when a file exports non-react components. The problem is that Clojure by default exports any non private var, and Squint in particular seems to export any variable (this may be a bug).

Example warning from Vite:
```js
12:03:27 AM [vite] hmr invalidate /src/main.jsx Could not Fast Refresh. Learn more at https://github.com/vitejs/vite-plugin-react/tree/main/packages/plugin-react#consistent-components-exports
```

**Describe the solution you'd like**

Only export vars with metadata `:export`. This would be similar to what [shadow-cljs](https://shadow-cljs.github.io/docs/UsersGuide.html#_working_with_optimizations) provides for external modules.

**Comment by borkdude** [comment]
Thanks for the issue. By design squint exports any non-private var, that is intentional.
But you're right that squint also exports private vars, created with `defn-`. This is something we can fix. Would that be a solution to your problem?

**Comment by aleedmanath** [comment]
The only drawback is doing `def ^:private` for any variable.

But perhaps one should be setting up HMR with Clojurescript macros rather than Vite's default React support.

**Comment by borkdude** [comment]
I can't change the current behavior because the normal behavior of ClojureScript is that you can access vars from other ClojureScript namespaces without adding `^:export`. 

But fixing `defn-` would be good in all cases, so let's start with that.

I'm not sure what you mean by HMR with CLJS macros.

**Comment by borkdude** [comment]
Released in 0.6.89
